profiling-ffi/src/profiles.rs

```diff
@@ -128,9 +131,11 @@ pub struct Period<'a> {
 #[derive(Copy, Clone, Default)]
 pub struct Label<'a> {
     pub key: CharSlice<'a>,
+    pub key_id: ManagedStringId,
 
     /// At most one of the following must be present
     pub str: CharSlice<'a>,
+    pub str_id: ManagedStringId,
```

Comment:
Something I wondered about while I was trying this out for the Python profiler: I assume the caller would need to check that this is nonzero before calling unintern on it, right? Or in any other case where a string might be optional. Maybe unintern should do nothing with managed strings with value 0, so it's safe to just always call unintern when you're done with a managed string ID?

Reply:
Makes sense, added in 6a658b4ea76407a3ff74a765ccd67b82634e48a6

Reply:
What about id=0 always being the empty string? In many string and string table things I've messed around with in the past, this has been a desirable behavior.